src/include/txn_inline.h

```diff
@@ -631,7 +631,14 @@ __wt_txn_tw_stop_visible(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
 static inline bool
 __wt_txn_tw_start_visible(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
 {
-    return ((WT_TIME_WINDOW_HAS_STOP(tw) || !tw->prepare) &&
+    /*
+     * Check the prepared flag if there is no stop time point or the start and stop time points are
+     * from the same transaction.
+     */
+    return (((WT_TIME_WINDOW_HAS_STOP(tw) &&
```

Comment:
Do a similar change in the function `__wt_txn_tw_start_visible_all` also.

Reply:
good point